muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/chat/dao/UserChatSettingsDAO.java

```diff
@@ -0,0 +1,68 @@
+package fi.otavanopisto.muikku.plugins.chat.dao;
+
+import java.util.List;
+
+import javax.persistence.EntityManager;
+import javax.persistence.TypedQuery;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Root;
+
+import fi.otavanopisto.muikku.plugins.CorePluginsDAO;
+import fi.otavanopisto.muikku.plugins.chat.model.UserChatSettings;
+import fi.otavanopisto.muikku.plugins.chat.model.UserChatSettings_;
+import fi.otavanopisto.muikku.plugins.chat.model.UserChatVisibility;
+import fi.otavanopisto.muikku.plugins.forum.model.ForumMessage;
+import fi.otavanopisto.muikku.schooldata.SchoolDataIdentifier;
+
+
+public class UserChatSettingsDAO extends CorePluginsDAO<UserChatSettings> {
+	
+  private static final long serialVersionUID = -7830619828801454118L;
+
+  public UserChatSettings create(String userIdentifier, UserChatVisibility userChatVisibility) {
+    UserChatSettings settings = new UserChatSettings(userIdentifier, userChatVisibility);
+    getEntityManager().persist(settings);
+    return settings;
+  }
+  
+  public List<UserChatSettings> listAll(String userIdentifier, UserChatVisibility userChatVisibility) {
+	EntityManager entityManager = getEntityManager(); 
+	    
+	CriteriaBuilder criteriaBuilder = entityManager.getCriteriaBuilder();
+	CriteriaQuery<UserChatSettings> criteria = criteriaBuilder.createQuery(UserChatSettings.class);
+	Root<UserChatSettings> root = criteria.from(UserChatSettings.class);
+	criteria.select(root);
+	
+	TypedQuery<UserChatSettings> query = entityManager.createQuery(criteria);
+	
+	return query.getResultList();
+  }
+  
+  public UserChatSettings findByUser(SchoolDataIdentifier userIdentifier) {
+    EntityManager entityManager = getEntityManager(); 
+    
+    CriteriaBuilder criteriaBuilder = entityManager.getCriteriaBuilder();
+    CriteriaQuery<UserChatSettings> criteria = criteriaBuilder.createQuery(UserChatSettings.class);
+    Root<UserChatSettings> root = criteria.from(UserChatSettings.class);
+    criteria.select(root);
+    criteria.where(
+        criteriaBuilder.equal(root.get(UserChatSettings_.userIdentifier), userIdentifier.toId())
+    );
+    
+    return getSingleResult(entityManager.createQuery(criteria));
+  }
+  
+  @Override
+  public void delete(UserChatSettings messageSignature) {
```

Comment:
Unused method. `delete` is another method all DAO classes have through inheritance and even if it has been overridden here, the code simply calls the underlying class so it's not even overriding anything at all.

Reply:
IIRC delete is private or protected though so overriding is bumping the visibility. That was the original intent at least so that delete would only be public where designed so.

Variable name could be more suited to the occation though.

Reply:
Yeah, sorry, totally forgot delete was protected and has to be made public. In that case, changing messageSignature to userChatSettings is enough.